pkg/apis/core/validation/project.go

```diff
@@ -60,10 +62,18 @@ func ValidateProjectUpdate(newProject, oldProject *core.Project) field.ErrorList
 func ValidateProjectSpec(projectSpec *core.ProjectSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	reservedNamespaceNames := []string{core.GardenerSeedLeaseNamespace, core.GardenerShootIssuerNamespace, core.GardenerSystemPublicNamespace}
-	if projectSpec.Namespace != nil && slices.Contains(reservedNamespaceNames, *projectSpec.Namespace) {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("namespace"), projectSpec.Namespace, fmt.Sprintf("Project namespaces %q are reserved by Gardener", reservedNamespaceNames)))
+	if projectSpec.Namespace != nil {
+		reservedNamespaceNames := []string{core.GardenerSeedLeaseNamespace, core.GardenerShootIssuerNamespace, core.GardenerSystemPublicNamespace}
+		if slices.Contains(reservedNamespaceNames, *projectSpec.Namespace) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("namespace"), *projectSpec.Namespace, fmt.Sprintf("Project namespaces [%s] are reserved by Gardener", strings.Join(reservedNamespaceNames, ", "))))
+			return allErrs
```

Comment:
Previously we didn't return here? Is this intended?

Reply:
Yes. I suggested to early exit to be added in https://github.com/gardener/gardener/pull/12784#discussion_r2290830861. The motivation was to for reserved namespace name to only return the concrete error (`Project namespaces [...] are reserved by Gardener`) and not to return in addition the error that the namespace does not start with a `garden-` prefix (`must start with garden-`).